src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/MissingJavadocPackageCheck.java

```diff
@@ -107,24 +107,31 @@ public void visitToken(DetailAST ast) {
      * @return true if there is javadoc, false otherwise
      */
     private static boolean hasJavadoc(DetailAST ast) {
-        final boolean hasBefore = isJavadocComment(ast.getPreviousSibling());
-        // need to go 3 levels down the tree for annotation case
-        final boolean hasWithAnnotation = Optional.of(ast.getFirstChild())
+        final boolean hasBefore = ast.getPreviousSibling() != null
+            && isJavadoc(ast.getPreviousSibling());
+        final Optional<DetailAST> firstAnnotationChild = Optional.of(ast.getFirstChild())
             .map(DetailAST::getFirstChild)
-            .map(DetailAST::getFirstChild)
-            .map(MissingJavadocPackageCheck::isJavadocComment)
-            .orElse(false);
+            .map(DetailAST::getFirstChild);
+        boolean hasWithAnnotation = false;
+        if (firstAnnotationChild.isPresent()) {
```

Comment:
we do not need to execute this IF block if `hasBefore` is true. Please refactor the code.

Reply:
done